test/functional/testlib.bash

```diff
@@ -0,0 +1,502 @@
+		echo -e "\\t-v\\tThe version for the bundle, if non selected version 10 will be used"
+		echo -e "\\t-d\\tComma-separated list of directories to be included in the bundle"
+		echo -e "\\t-f\\tComma-separated list of files to be created and included in the bundle"
+		echo -e "\\t-l\\tComma-separated list of symlinks to be created and included in the bundle,"
+		echo "Notes:"
+		echo -e "\\t- if no option is selected, a minimal bundle will be created with only one directory"
+		echo -e "\\t- for every symlink created a related file will be created and added to the bundle as well"
+		echo -e "\\t- if the '-f' or '-l' options are used, and the directories where the files live don't exist,"
+		echo -e "\\t  they will be automatically created and added to the bundle for each file"
+		echo -e "\\nExample of usage:\\n"
+		echo -e "\\tThe following command will create a bundle named 'test-bundle', which will include three directories,"
+		echo -e "\\tfour files, and one symlink (they will be added to the bundle's manifest), all these resources will also"
+		echo -e "\\tbe tarred. The bundle's manifest will be added to the MoM."
+		echo -e "\\n\\tcreate_bundle -n test-bundle -f /usr/bin/test-1,/usr/bin/test-2,/etc/systemd/test-3 -l /etc/test-link my_test_env"
+		echo -e "\\n"
+		
+	}
+	
+	local OPTIND
+	local opt
+	local dir_list
+	local file_list
+	local link_list
+	local version
+	local bundle_name
+	local env_name
+	local files_path
+	local version_path
+	local manifest
+	local local_bundle=false
+
+	set -f  # turn off globbing
+	while getopts :v:d:f:l:n:L opt; do
+		case "$opt" in
+			d)	IFS=,
+				read -r -a dir_list <<< "$OPTARG" ;;
+			f)	IFS=,
+				read -r -a file_list <<< "$OPTARG" ;;
+			l)	IFS=,
+				read -r -a link_list <<< "$OPTARG" ;;
+			n)	bundle_name="$OPTARG" ;;
+			v)	version="$OPTARG" ;;
+			L)	local_bundle=true ;;
+			*)	cb_usage
+				return ;;
+		esac
+	done
+	set +f  # turn globbing back on
+	env_name=${@:$OPTIND:1}
+	
+	# set default values
+	bundle_name=${bundle_name:-$(generate_random_name test-bundle-)}
+	version=${version:-10}
+	if [ -z "$dir_list" ] && [ -z "$file_list" ] && [ -z "$link_list" ] ; then
+		# if nothing was specified to be created, at least create
+		# one directory which is the bare minimum for a bundle
+		dir_list=(/usr/bin)
+	fi
+	# all bundles should include their own tracking file, so append it to the
+	# list of files to be created in the bundle
+	file_list+=(/usr/share/clear/bundles/"$bundle_name")
+	
+	# get useful paths
+	if [ "$(validate_path "$env_name")" ]; then
+		echo "Please specify a valid environment"
+		return 1
+	fi
+	version_path="$env_name"/web-dir/"$version"
+	files_path="$version_path"/files
+	target_path="$env_name"/target-dir
+
+	# 1) create the initial manifest
+	manifest=$(create_manifest "$version_path" "$bundle_name")
+	echo "Manifest -> $manifest"  # TODO(castulo): remove this msg when finish development
```

Comment:
Could create a "debug" function which outputs these messages iff an environment variable is set

Reply:
that is a good idea